data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostSubscribeToBeaconCommitteeSubnet.java

```diff
@@ -13,44 +13,100 @@
+      COMMITTEE_SUBSCRIPTION_REQUEST_TYPE =
+          DeserializableTypeDefinition.object(CommitteeSubscriptionData.class)
+              .name("CommitteeSubscriptionData")
+              .initializer(CommitteeSubscriptionData::new)
+              .withField(
+                  "validator_index",
+                  INTEGER_TYPE,
+                  CommitteeSubscriptionData::getValidatorIndex,
+                  CommitteeSubscriptionData::setValidatorIndex)
+              .withField(
+                  "committee_index",
+                  INTEGER_TYPE,
+                  CommitteeSubscriptionData::getCommitteeIndex,
+                  CommitteeSubscriptionData::setCommitteeIndex)
+              .withField(
+                  "committees_at_slot",
+                  UINT64_TYPE,
+                  CommitteeSubscriptionData::getCommitteesAtSlot,
+                  CommitteeSubscriptionData::setCommitteesAtSlot)
+              .withField(
+                  "slot",
+                  UINT64_TYPE,
+                  CommitteeSubscriptionData::getSlot,
+                  CommitteeSubscriptionData::setSlot)
+              .withField(
+                  "is_aggregator",
+                  BOOLEAN_TYPE,
+                  CommitteeSubscriptionData::isAggregator,
+                  CommitteeSubscriptionData::setAggregator)
+              .build();
+
+  public PostSubscribeToBeaconCommitteeSubnet(final DataProvider dataProvider) {
+    this(dataProvider.getValidatorDataProvider());
   }
 
-  public PostSubscribeToBeaconCommitteeSubnet(
-      final ValidatorDataProvider provider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
+  public PostSubscribeToBeaconCommitteeSubnet(final ValidatorDataProvider provider) {
+    super(
+        EndpointMetadata.post(ROUTE)
+            .operationId("postSubscribeToBeaconCommitteeSubnet")
+            .summary("Subscribe to a committee subnet")
+            .description(
+                "After Beacon node receives this request, search using discv5 for peers related to this subnet and replace current peers with those ones if necessary If validator is_aggregator, beacon node must:\n"
+                    + "- announce subnet topic subscription on gossipsub\n"
+                    + "- aggregate attestations received on that subnet\n")
+            .tags(TAG_VALIDATOR, TAG_VALIDATOR_REQUIRED)
+            .requestBodyType(
+                DeserializableTypeDefinition.listOf(COMMITTEE_SUBSCRIPTION_REQUEST_TYPE))
+            .response(
+                HttpStatusCodes.SC_OK,
+                "Slot signature is valid and beacon node has prepared the attestation subnet. Note that, there is no guarantee the node will find peers for the subnet")
+            .response(
+                SC_SERVICE_UNAVAILABLE,
+                "Beacon node is currently syncing and not serving request on that endpoint",
+                HTTP_ERROR_RESPONSE_TYPE) // TODO remove?
```

Comment:
service unavailable will definitely use the response, potentially could make this a 'withServiceUnavailableResponse' and have it all defined for you...